src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/includeClassifier.js

```diff
@@ -0,0 +1,52 @@
+import Url from 'url-parse';
+
+const supportedTypes = {
+  indepthtoolkit: 'idt1',
+  idt2: 'idt2',
+  include: 'vj',
+  'news/special': 'vj',
+  'market-data': 'vj',
+  'smallprox/include': 'vj',
+};
+
+const ampSupported = href => {
+  // An amp-image query parameter on the include path indicates an AMP version of the include is available
+  const hasAmpImageQueryString = new Url(href, true).query['amp-image'];
+  return !!hasAmpImageQueryString;
+};
+
+const includeClassifier = ({ href, pathname }) => {
+  // This determines if the href has a leading '/'
+  const hrefTypePostion = () => (href.indexOf('/') === 0 ? 1 : 0);
+
+  // This checks if the supportedType is in the correct position of the href
+  const hrefIsSupported = () => supportedType =>
+    href.startsWith(supportedType, hrefTypePostion());
+
+  // This extracts the type from the href
+  const typeExtraction = Object.keys(supportedTypes).find(
+    hrefIsSupported(href),
+  );
+
+  // This determines if the type is supported and returns the include type name
+  const includeType = supportedTypes[typeExtraction];
+
+  const pathnameIsAmp = pathname.endsWith('.amp');
+
+  const platform = pathnameIsAmp ? 'amp' : 'canonical';
+  const classification = includeType
+    ? `${includeType}-${platform}`
+    : 'not-supported';
+
+  if (!pathnameIsAmp || (!ampSupported(href) && classification !== 'vj-amp')) {
+    return { includeType, classification };
+  }
+
+  if (ampSupported(href)) {
+    return { includeType, classification: 'vj-include-supports-amp' };
+  }
```

Comment:
I find this a bit tricky to read as we have a lot of negative conditionals above, maybe move this if statement up so you can test for:
`if(includeType === 'vj' && ampSupported(href))`

Then you can positively test for each condition and return the correct classfication object. In my head I when I read `if(includeType === 'vj' && ampSupported(href))` I hear "when include type is VJ and it supports amp, return this" as opposed to "when the url is not an amp page, the include does not support amp and it has the classification 'vj-amp', return this" for `!pathnameIsAmp || (!ampSupported(href) && classification !== 'vj-amp')`.

Sorry for the waffle! Just wanted to get across my reasoning for avoiding negations and that's the best I've got 😉

Reply:
Thanks for addressing the reads really well now 💯